tests/e2e/mnist_raycluster_sdk_auth_test.go

```diff
@@ -0,0 +1,439 @@
+			Parallelism:  Ptr(int32(1)),
+			BackoffLimit: Ptr(int32(0)),
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "test",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: config.Name,
+									},
+								},
+							},
+						},
+						{
+							Name: "codeflare-sdk",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+							Name: "workdir",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name: "test",
+							// FIXME: switch to base Python image once the dependency on OpenShift CLI is removed
+							// See https://github.com/project-codeflare/codeflare-sdk/pull/146
+							Image: "quay.io/opendatahub/notebooks:jupyter-minimal-ubi8-python-3.8-4c8f26e",
+							Env: []corev1.EnvVar{
+								{Name: "PYTHONUSERBASE", Value: "/workdir"},
+								{Name: "RAY_IMAGE", Value: GetRayImage()},
+								{Name: "OAUTH_LOGIN", Value: "false"},
+							},
+							Command: []string{
+								"/bin/sh", "-c",
+								"while [ ! -f /codeflare-sdk/pyproject.toml ]; do sleep 1; done; " +
+									"cp /test/* . && chmod +x install-codeflare-sdk.sh && ./install-codeflare-sdk.sh && python mnist_raycluster_sdk_auth.py " + namespace.Name,
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "test",
+									MountPath: "/test",
+								},
+								{
+									Name:      "codeflare-sdk",
+									MountPath: "/codeflare-sdk",
+								},
+								{
+									Name:      "workdir",
+									MountPath: "/workdir",
+								},
+							},
+							WorkingDir: "/workdir",
+							SecurityContext: &corev1.SecurityContext{
+								AllowPrivilegeEscalation: Ptr(false),
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: "RuntimeDefault",
+								},
+								Capabilities: &corev1.Capabilities{
+									Drop: []corev1.Capability{"ALL"},
+								},
+								RunAsNonRoot: Ptr(true),
+							},
+						},
+					},
+					RestartPolicy:      corev1.RestartPolicyNever,
+					ServiceAccountName: sa.Name,
+				},
+			},
+		},
+	}
+	job, err := test.Client().Core().BatchV1().Jobs(namespace.Name).Create(test.Ctx(), job, metav1.CreateOptions{})
+	test.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created Job %s/%s successfully", job.Namespace, job.Name)
+
+	// Setup the codeflare-sdk inside the pod associated to the created job
+	SetupCodeflareSDKInsidePod(test, namespace, job.Name)
+
+	test.T().Logf("Checking pod logs contains insecure requests warning")
+	// Assert the job pod as warning of insecurerequests
+	assertPodLogs(test, namespace, job.Name, "InsecureRequestWarning: Unverified HTTPS request is being made to host", 3*time.Minute)
```

Comment:
I would expect SDK to provide some information about the issue, not just `InsecureRequestWarning` which is not relevant to the actual problem.

Reply:
Thinking whether we should explicitly check for the dashboard/API availability. This line verifies that SDK returns a warning, possibly as part of `cluster.wait_ready()` call. However implementation of this call can change making this test incorrect.

Would it have sense to add specific REST call in Python script invoking Dashboard/Jobs API, making sure those call returns correct response code for unauthenticated user?

Reply:
Yes I think its make sense to have check to correct unauthorised response code and message

Reply:
@sutaakar @Bobbins228 @KPostOffice  I tried to Submit Ray job in multiple ways for testing unauthenticated scenario but getting same below response error as in UI as login should required to connect with Ray dashboard. 

```
resp=requests.post(dashboard_url+"/api/jobs/",
                   json={ "entrypoint":"python mnist.py",
    "runtime_env":{"working_dir": "workdir","pip": "requirements.txt"}})

client = RayJobClient(dashboard_url)
submission_id = client.submit_job(
    entrypoint="python mnist.py",
    runtime_env={"working_dir": "./","pip": "mnist_pip_requirements.txt"},
)

 An unexpected error occurred: HTTPSConnectionPool(host='ray-dashboard-test.com', port=443): Max retries exceeded with url: /api/jobs/ (Caused by SSLError(SSLCertVerificationError(1, [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1007)')))
```

I think for Authentication tests we can go with current implementation approach with cluster token and verify Ray Job submission with authentication and  for  non authentication tests as we don't get exact error saying that Not authorised to submit Job, either we can have negative check incase of submitting job to ray dashboard which is not allowing to submit by validating response code other than 200 or skipping this test scenario .

Reply:
@Srihari1192 Have you tried to disable certificate verification?
https://stackoverflow.com/questions/15445981/how-do-i-disable-the-security-certificate-check-in-python-requests

Reply:
@sutaakar By disabling certificate verification  able to test this will update the PR